clang/lib/AST/JSONNodeDumper.cpp

```diff
@@ -1724,6 +1724,8 @@ void JSONNodeDumper::VisitTemplateExpansionTemplateArgument(
 void JSONNodeDumper::VisitExpressionTemplateArgument(
     const TemplateArgument &TA) {
   JOS.attribute("isExpr", true);
+  if (TA.isCanonicalExpr())
+    JOS.attribute("isCanon", true);
```

Comment:
Any reason to not just do `isCanonical` instead?  `Canon` is already a word and sounds nonsensical.